velox/vector/tests/utils/VectorTestBase.h

```diff
@@ -48,7 +50,27 @@ void assertCopyableVector(const VectorPtr& vector);
 
 class VectorTestBase {
  protected:
-  VectorTestBase() = default;
+  VectorTestBase() {
```

Comment:
@xiaoxmeng  I'm not sure I understand this change. My understanding is that memory arbitrator is used to (1) arbitrate memory usage between tasks; (2) arbitrate memory usage between operators within a single task.

Most tests run one task at a time, hence, (2) is the main use case at the moment.

These test run tasks using code in Cursor.h/cpp which creates tasks using AssertQueryBuilder::readCursor or similar.

I'm seeing that QueryCtx::initPool uses memory::defaultMemoryManager().

l'm also seeing that memory::defaultMemoryManager() goes into MemoryManager::getInstance(options). This is where things get tricky for me. Looks like MemoryManager::getInstance() method creates an instance of MemoryManager on first class using specified options. On subsequent calls it ignores 'options' parameter and returns instance created earlier. Is this correct? Is this design intentional?

It seems that whoever calls MemoryManager::getInstance() first gets to decide what the settings will be... It seems quite fragile as it is hard to ensure that any particular place will be the first one... unless it is happening in main(). Am I missing something?

```
QueryCtx:

  void initPool(const std::string& queryId) {
    if (pool_ == nullptr) {
      pool_ = memory::defaultMemoryManager().addRootPool(
          QueryCtx::generatePoolName(queryId));
    }
  }


void OperatorTestBase::SetUp() {
  if (!isRegisteredVectorSerde()) {
    this->registerVectorSerde();
  }
  driverExecutor_ = std::make_unique<folly::CPUThreadPoolExecutor>(3);
  ioExecutor_ = std::make_unique<folly::IOThreadPoolExecutor>(3);
  allocator_ = std::make_shared<memory::MallocAllocator>(8L << 30);
  if (!asyncDataCache_) {
    asyncDataCache_ = cache::AsyncDataCache::create(allocator_.get());
    cache::AsyncDataCache::setInstance(asyncDataCache_.get());
  }
  memory::MemoryAllocator::setDefaultInstance(allocator_.get());
}

MemoryManager& MemoryManager::getInstance(const MemoryManagerOptions& options) {
  static MemoryManager manager{options};
  return manager;
}
```

Reply:
In addition, VectorTestBase is used by many tests which do not run tasks. These tests do not need any memory arbitrator. Perhaps, a more flexible design would be for VectorTestBase to stay as is and rely on the "caller" to setup default memory manager.

Reply:
> Perhaps, a more flexible design would be for VectorTestBase to stay as is and rely on the "caller" to setup default memory manager.

+1 on this proposal. Each test should explicitly enable arbitration so that the code paths are also deterministic. For example, we don't want HashJoin to spill if the test expectation is to test the memory limits of HashJoin.